lms/djangoapps/discussion_api/tests/utils.py

```diff
@@ -226,6 +226,15 @@ def register_get_user_retire_response(self, user, status=200, body=""):
             status=status
         )
 
+    def register_get_username_replacement_response(self, user, status=200, body=""):
+        assert httpretty.is_enabled(), 'httpretty must be enabled to mock calls.'
```

Comment:
I think you should use APITestCase (https://www.django-rest-framework.org/api-guide/testing/) instead of httppretty.  httppretty won't actually let you get to the underlying api

Reply:
i think in that case you should use http pretty to mock the underlying API

Reply:
oh wait i see this is mocking the underlying api. okay nevermind ignore my comment